### PR TITLE
Dropped ES.. Looks like most working

### PR DESCRIPTION
**Pull request by mharasic**

### CODE REVIEW

File: backend/api/search.py [hunk 1]
```diff
@@ -19,10 +18,8 @@
 def prepare_es_results(res):
     """ Internal function to prepare ElasticSearch search results to send as json.
     """
+    projects = [x for x in res[:]]
```
Comment (henderkes) [inline backend/api/search.py]: list(res) is easier

File: backend/api/search.py [hunk 2]
```diff
@@ -36,7 +33,7 @@ def prepare_es_results(res):
 def prepare_es_query(query):
     """ Internal function to prepare the ElasticSearch search query from a given json
     """
-    search_string = query.get('searchString', '')
+    search_string = query.get('searchString')
```
Comment (henderkes) [inline backend/api/search.py]: this needs to be checked for None later now

File: backend/api/search.py [hunk 3]
```diff
@@ -131,10 +76,10 @@ def search_es():
     query = request.get_json()
     request_json = prepare_es_query(query)
 
-    try:
-        projects = prepare_es_results(g.es.search(index="knexdb", body=request_json))
-    except RequestError as reqerr:
-        raise ApiException(str(reqerr), 400)
+    projects = g.projects.find(request_json, {'comments': 0})
+    projects = [x for x in projects[:]]
```
Comment (henderkes) [inline backend/api/search.py]: see above

File: backend/requirements.txt [hunk 4]
```diff
@@ -1,6 +1,4 @@
 flask
-pymongo
```
Comment (DubbleClick) [inline backend/requirements.txt]: pymongo must stay

File: backend/api/search.py [hunk 5]
```diff
@@ -185,7 +130,7 @@ def query_saved_search(id):
                 query = json.loads(cursearch['query'])
                 query['size'] = count
                 query['from'] = offset
-                projects = prepare_es_results(g.es.search(index="knexdb", body=query))
+                projects = prepare_es_results(g.projects.find(request_json, {'comments': 0}))
```
Comment (DubbleClick) [inline backend/api/search.py]: prepare_search results method is no longer needed

**Review by henderkes — DISMISSED** [review]
okay I'll work on this tomorrow, still needs a ton of changes

File: backend/api/search.py (hunk 5 above)
Comment (mharasic) [inline backend/api/search.py]: It is still needed somehow.... It transforms it, but I will refactor

File: backend/api/search.py (hunk 2 above)
Comment (mharasic) [reply backend/api/search.py]: Nope... Indexes are smart enough to omit empty strings :)

Comment (henderkes) [reply backend/api/search.py]: this will return None now instead of an empty string

Comment (henderkes) [reply backend/api/search.py]: well... it's not used anywhere, so why bother getting it in the first place?

Comment (mharasic) [reply backend/api/search.py]: It should go over the indixes, but can work with an empty string

**Review by henderkes — DISMISSED** [review]

**Review by henderkes — APPROVED** [review]

**Pull request merged**